### PR TITLE
Docker 101 - initial draft

### PR DESCRIPTION
**Pull request by jefferya**
Goal: discussion starter around documentation on the usage of Docker within DI

* attempts to outline the purpose, usage
* proposes a template for documenting a Docker GitHub repo (and maybe a DockerHub repo as well)
* some details may be wrong or there may be better ways to accomplish certain tasks

Warning: the Docker Compose sections are weak and show my relatively newness to Docker in general.

Thoughts / additions / subtractions welcomed,

### CODE REVIEW

**Comment by jefferya** [comment]
An example of the proposed template - partially complete:
https://github.com/ualbertalib/pushmi_pullyu/tree/issue_25_docker_image/docker

**Comment by cwant** [comment]
A rendered view of the content for *this* pull request can be seen here:

https://github.com/ualbertalib/Developer-Handbook/blob/Docker_101/Docker/README.md

File: Docker/README.md [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+
+
+## Maintenance/Creation Tasks ##
+
+### Creating from scratch  ###
+
+* Create a GitHub repo to contain the files
+* Start with [README.md template](#readme-template)
+* Populate the supporting Docker files (e.g, Dockerfile, docker-compose.yml)
+* Build the Docker image from `Dockerfile` definition
```
Comment (weiweishi) [inline Docker/README.md]: Maybe add the "how to build" command here?

File: Docker/README.md [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+* Test image
+* Push to https://hub.docker.com/r/ualibraries/
```
Comment (weiweishi) [inline Docker/README.md]: Maybe a separate section on how to push/update images on Docker hub (with proper tag if necessary)

File: Docker/README.md [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+
+## Maintenance/Creation Tasks ##
+
+### Creating from scratch  ###
+
+* Create a GitHub repo to contain the files
+* Start with [README.md template](#readme-template)
+* Populate the supporting Docker files (e.g, Dockerfile, docker-compose.yml)
+* Build the Docker image from `Dockerfile` definition
+* Test image
```
Comment (weiweishi) [inline Docker/README.md]: Add CI testing requirement here. See project document that was just approved from review yesterday: https://github.com/ualbertalib/di_docker_images/blob/master/README.md

File: Docker/README.md [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+
+## Special notes / warnings / gotchas ##
+
+
+
+## Future considerations ##
+
+* Consider [configuration of automated builds on Docker Hub](https://docs.docker.com/docker-hub/builds/)
+
```
Comment (weiweishi) [inline Docker/README.md]: Maybe add Docker Swarm/Atomic as mentioned here: https://github.com/ualbertalib/di_docker_images/issues/10 and https://github.com/ualbertalib/di_docker_images/issues/11 as future consideration

File: Docker/README.md (hunk 2 above)
Comment (murny) [inline Docker/README.md]: Side point is how does someone get access to this ualibraries dockerhub?  To push/update images to it? Do we have to go bug Sam or Alex? etc.

File: Docker/README.md [hunk 5]
```diff
@@ -0,0 +1,278 @@
+# Docker 101 #
+
+## What is Docker? ##
```
Comment (murny) [inline Docker/README.md]: Seen someone else do this on our team. So very small nitpick but you don't need the double `#` hash marks on both sides? Only one side is needed and is considered proper markdown

```
# Header
```
vs
```
# Header #
```

Official markdown syntax: http://daringfireball.net/projects/markdown/syntax#header
Github official markdown syntax: https://guides.github.com/features/mastering-markdown/#syntax

Comment (jefferya) [inline Docker/README.md]: I'll change to terse version as it looks more prevalent in other work.

File: Docker/README.md (hunk 2 above)
Comment (jefferya) [inline Docker/README.md]: Not sure of the Docker Hub process either.

**Comment by murny** [comment]
I'm going to merge this today unless anyone else has more feedback/changes to request?

**Comment by pbinkley** [comment]
:+1:

**Pull request merged**